ScoutSuite/providers/gcp/resources/gce/instances.py

```diff
@@ -0,0 +1,73 @@
+from ScoutSuite.providers.gcp.resources.resources import GCPCompositeResources
+from ScoutSuite.providers.gcp.resources.gce.instance_disks import InstanceDisks
+from ScoutSuite.providers.utils import get_non_provider_id
+
+class Instances(GCPCompositeResources):
+    _children = [ 
+        (InstanceDisks, 'disks')
+    ]
+
+    def __init__(self, gcp_facade, project_id, zone):
+        self.gcp_facade = gcp_facade
+        self.project_id = project_id
+        self.zone = zone
+
+    async def fetch_all(self):
+        raw_instances = await self.gcp_facade.gce.get_instances(self.project_id, self.zone)
+        for raw_instance in raw_instances:
+            name, instance = await self._parse_instance(raw_instance)
+            self[name] = instance
+  
+    async def _parse_instance(self, raw_instance):
+        instance_dict = {}
+        instance_dict['id'] = get_non_provider_id(raw_instance['name'])
+        instance_dict['project_id'] = self.project_id
+        instance_dict['name'] = raw_instance['name']
+        instance_dict['description'] = self._get_description(raw_instance)
+        instance_dict['creation_timestamp'] = raw_instance['creationTimestamp']
+        instance_dict['zone'] = raw_instance['zone'].split('/')[-1]
+        instance_dict['tags'] = raw_instance['tags']
+        instance_dict['status'] = raw_instance['status']
+        instance_dict['zone_url_'] = raw_instance['zone']
+        instance_dict['network_interfaces'] = raw_instance['networkInterfaces']
+        instance_dict['service_accounts'] = raw_instance['serviceAccounts']
+        instance_dict['deletion_protection_enabled'] = raw_instance['deletionProtection']
+        instance_dict['block_project_ssh_keys_enabled'] = self._is_block_project_ssh_keys_enabled(raw_instance)
+        instance_dict['oslogin_enabled'] = await self._is_oslogin_enabled(raw_instance)
+        instance_dict['ip_forwarding_enabled'] = raw_instance['canIpForward']
+        instance_dict['serial_port_enabled'] = self._is_serial_port_enabled(raw_instance)
+        instance_dict['has_full_access_cloud_apis'] = self._has_full_access_to_all_cloud_apis(raw_instance)
+        instance_dict['disks'] = InstanceDisks(raw_instance)
+        return instance_dict['id'], instance_dict
+
+    def _get_description(self, raw_instance):
+        description = raw_instance.get('description')   
+        return description if description else 'N/A'
+
+    def _is_block_project_ssh_keys_enabled(self, raw_instance):
+        metadata = self._metadata_to_dict(raw_instance['metadata'])
+        return metadata.get('block-project-ssh-keys') == 'true'
+
+    def _metadata_to_dict(self, metadata):
+        return dict((item['key'], item['value']) for item in metadata['items']) if 'items' in metadata else {}
+
+    async def _get_common_instance_metadata_dict(self):
```

Comment:
This method should be moved to GCE facade and those metadata should get attached to their instance objects when calling `get_instances` in order to maintain the separation of concerns principle (between the fetching and parsing parts) and also avoid performance issues (in this case, since the parsing is `async` because it does some fetching, the `for loop` in `fetch_all` is not really efficient and could lead to poor performances).
Although, regarding our schedule, I don't mind (do you @Aboisier ?) if you simply add a `TODO` to move forward and merge those PRs so we can unify a last time our refactoring.

Reply:
It would be ideal is if was fixed before merging, but I understand we're running out of time now. :/

Reply:
Good idead! I wasn't a fan of having an `async` parsing method.